### PR TITLE
Fix the gemini data manager, data tables and wrapper tests

### PR DESCRIPTION
**Pull request by wm75**
This changeset should make the gemini data manager fully functional
and eliminate the need of any manual patching of the installed
gemini data files.
It also introduces a new gemini_versioned_databases data table that
allows gemini tool wrappers to filter the installed gemini
annotation files for those they can actually work with.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

**Comment by wm75** [comment]
Ah, this breaks all tools that do not actually require the data files. I'll fix it tomorrow ...

**Comment by wm75** [comment]
Going to fix the failing tests ...

**Comment by wm75** [comment]
Some tests still depend on a handful of input databases that the `build-gemini-testdata.sh` script does not know how to build. Working on it ...

File: data_managers/data_manager_gemini_database_downloader/data_manager/data_manager_gemini_download.xml [hunk 1]
```diff
@@ -1,7 +1,11 @@
-<tool id="data_manager_gemini_download" name="GEMINI Download" version="0.19.1" tool_type="manage_data">
+<tool id="data_manager_gemini_download" name="GEMINI Download" version="@VERSION@.1" tool_type="manage_data">
     <description>Download a new database</description>
+    <macros>
+        <token name="@VERSION@">0.18.1</token>
+        <token name="@DB_VERSION@">181</token>
+    </macros>
     <requirements>
-        <requirement type="package" version="0.19.1">gemini</requirement>
+        <requirement type="package" version="0.18.1">gemini</requirement>
```
Comment (bgruening) [inline data_managers/data_manager_gemini_database_downloader/data_manager/data_manager_gemini_download.xml]: Here you can use @VERSION@ ...

Comment (wm75) [reply data_managers/data_manager_gemini_database_downloader/data_manager/data_manager_gemini_download.xml]: Thanks. Fixed in the latest commit

**Comment by wm75** [comment]
@bgruening I think that's how far I'd take this. The test coverage of the tools is still far from ideal, but, at least, gemini can now be tested at all.
After this gets merged I can open a new PR that upgrades gemini to version 0.20.1. I'll add more tool tests there.

File: tools/gemini/gemini_annotate.xml [hunk 2]
```diff
@@ -14,19 +14,19 @@
...
-            -a $a.a_selector
-            #if $a.a_selector == 'extract':
-                -t $a.column_type
-                -e $a.column_extracts
-                -o $a.operation
-            #end if
-            $region_only
-            "${ infile }"
-            > "${ outfile }"
+    cp "$infile" "$outfile" &&
```
Comment (bgruening) [inline tools/gemini/gemini_annotate.xml]: single quote

File: tools/gemini/gemini_annotate.xml [hunk 3]
```diff
@@ -14,19 +14,19 @@
...
+
+    gemini @BINARY@
+        -f $tabixed_file
+        -c $column_name
```
Comment (bgruening) [inline tools/gemini/gemini_annotate.xml]: quotes

**Comment by bgruening** [comment]
Super cool, just two small comments.

**Comment by bgruening** [comment]
Merci!

**Pull request merged**